### DISCUSSION TITLE
Speed limits, street names in GPX navigation track

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#404] [Q&A] [answered]
The top panel current street (if set) is shown at navigation by direct design.
The top panel current street (if set) is supported by a .kurviger file import.
The top panel current street is not supported by a gpx navtrack import.

Suggestion.
Add the current street in the gpx navtrkpt cmt.

Example.
![image](https://github.com/devemux86/cruiser/assets/54167201/b06ce508-b379-4dce-97a2-23e81e60e4cd)
[Current_street_navtrk.gpx.txt](https://github.com/devemux86/cruiser/files/14235912/Current_street_navtrk.gpx.txt)

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the suggestion.

Navigation GPX format currently does not store road details, like speed limits or street names.
We could also include such road details in the navigation GPX.

(they are already included in the JSON compatible formats)

**Comment by devemux86** [comment]
Speed limits and street names will also be saved in the (offline) GPX navigation track.

So it stores the same content as the (offline) JSON formats.

**Reply by umpfmpf** [reply]
So it is also visible during navigation?

**Reply by devemux86** [reply]
@umpfmpf correctly, all route data for navigation, just like json or kurviger files.

**Reply by WolfgangZ42** [reply]
> all route data for **navigation**, just like json or kurviger files

If it's in the BRouter data (offline available on my phone) you could present in "follow location" as well - unlike the kurviger files. Right?

**Reply by devemux86** [reply]
@WolfgangZ42 what data do you mean?

BRouter (offline) with the new reader can now provide much more data, such as the new speed limits:
- https://github.com/devemux86/cruiser/discussions/441

Usually such data is provided in the calculated routes which are used in navigation.

Follow location mode centers the map on GPS location, while the map can also display a route or track to follow.

**Reply by WolfgangZ42** [reply]
> Usually such data is provided in the **calculated** routes

In follow mode you know where I am. And I thought you probably could read the current speed limit from the BRouter data. But you can use the latter one only for routing? Sorry for my wrong idea ...

**Reply by Ciutnik** [reply]
> BRouter (offline) with the new reader can now provide much more data

Maybe the street names too? That would be great... 😳

**Reply by devemux86** [reply]
@WolfgangZ42 now I understand what you mean, this could be possible. :slightly_smiling_face:

**Reply by devemux86** [reply]
@Ciutnik unfortunately BRouter does not store street names it its data.

We would all like to change that. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> BRouter does not store street names it its data

Oh, that's a pity. I hoped something had changed. Well, we still have to wait. Maybe someday we'll see it.

**Reply by devemux86** [reply]
BRouter (offline) provides a lot of data. I created a new topic to discuss what we would like to display:
- https://github.com/devemux86/cruiser/discussions/442

Now with the new Cruiser reader of BRouter data, this is easy. :slightly_smiling_face:

**Comment by devemux86** [comment]
Speed limits, street names in GPX navigation track are available for testing in:
- https://github.com/devemux86/cruiser/discussions/451

**Comment by devemux86** [comment, marked as answer]
Speed limits, street names in GPX navigation track are available in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by bios4** [reply]
How do I get these details in a/the GPX track @devemux86?

I loaded a test route and exported in GPX format, but could neither find speed limits nor street names in the exported GPX file.
Just the "default" syntax:

`<trkpt lat="XX.XXXXX" lon="YY.YYYYY">
<ele>362.25</ele>
<time>2024-03-18T15:59:01Z</time>
</trkpt>`

Or is this applicable/available only when NOT using BRouter?
Maybe this should be explicitly mentioned then...

**Reply by devemux86** [reply]
@bios4 they are included in the GPX **navigation** format (not simple GPX).
Check the "Navigation" in the export dialog.

The export includes the extra data that is available in the calculated or imported route.
e.g. BRouter (internal) now provides speed limits (still no street names).
Imported curvy routes should provide speed limits and street names.

**Reply by bios4** [reply]
> Check the "Navigation" in the export dialog.

The checkbox WAS checked in my export!

> BRouter (internal) now provides speed limits (still no street names).

OK then it is a limitation of BRouter.
Speed limits are there (where available from the routing data, not for all entries), but no street names...

Thanks!
